openedx/core/djangoapps/user_api/validation/tests/test_views.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for an API endpoint for client-side user data validation.
+"""
+
+import ddt
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+
+from openedx.core.djangoapps.user_api.accounts import (
+    EMAIL_CONFLICT_MSG, EMAIL_MAX_LENGTH, EMAIL_MIN_LENGTH,
+    PASSWORD_MAX_LENGTH, PASSWORD_MIN_LENGTH,
+    USERNAME_CONFLICT_MSG, USERNAME_MAX_LENGTH, USERNAME_MIN_LENGTH
+)
+from openedx.core.djangoapps.user_api.accounts.tests.testutils import (
+    VALID_EMAILS, VALID_PASSWORDS, VALID_USERNAMES,
+    INVALID_EMAILS, INVALID_PASSWORDS, INVALID_USERNAMES
+)
+from openedx.core.lib.api import test_utils
+
+
+@ddt.ddt
+class RegistrationValidationViewTests(test_utils.ApiTestCase):
+    """
+    Tests for validity of user data in registration forms.
+    """
+
+    def setUp(self):
+        super(RegistrationValidationViewTests, self).setUp()
+        self.endpoint_name = 'registration_validation'
+        self.path = reverse(self.endpoint_name)
+
+    def get_validation_decision(self, data):
+        response = self.client.get(self.path, data)
+        return response.data.get('validation_decisions', {})
+
+    def test_no_decision_for_empty_request(self):
+        validation_decisions = self.get_validation_decision({})
+        self.assertEqual(validation_decisions, {})
+
+    def test_no_decision_for_invalid_request(self):
+        validation_decisions = self.get_validation_decision({
+            'invalid_field': 'random_user_data'
+        })
+        self.assertEqual(validation_decisions, {})
+
+    @ddt.data(
+        ['email', VALID_EMAILS],
+        ['password', VALID_PASSWORDS],
+        ['username', VALID_USERNAMES]
+    )
+    @ddt.unpack
+    def test_positive_validation_decision(self, form_field_name, user_data):
+        """
+        Test if {0} as any item in {1} gives a positive validation decision.
+        """
+        for data in user_data:
+            self.assertEqual(
+                self.get_validation_decision({form_field_name: data}),
+                {form_field_name: ''}
+            )
+
+    @ddt.data(
+        # Skip None type for invalidity checks.
+        ['email', INVALID_EMAILS[1:]],
+        ['password', INVALID_PASSWORDS[1:]],
+        ['username', INVALID_USERNAMES[1:]]
+    )
+    @ddt.unpack
+    def test_negative_validation_decision(self, form_field_name, user_data):
+        """
+        Test if {0} as any item in {1} gives a negative validation decision.
+        """
+        for data in user_data:
+            self.assertNotEqual(
+                self.get_validation_decision({form_field_name: data}),
+                {form_field_name: ''}
+            )
```

Comment:
Does `ddt` provide something here for me to avoid the for loop? How would I iterate over each invalid email with `email` as the `form_field_name` separately, and so on with just `ddt`?